### PR TITLE
Custom registration form error messages

### PR DESCRIPTION
**Pull request by inaod568**
Added custom messages. Currently not showing up, for them to actually be visible we need to add `novalidate` to the html form. 

- [x] Tests passing
- [x] Black formatting
- [ - ] Migrations for any changes to the database schema
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

**Comment by ewels** [comment]
Nice! Much of this validation we can also do in the front-end if we want, both with native browser controls and maybe some simple javascript. Can be nice to get immediate validation errors before submitting the form.

Any chance the things like username / password lengths etc could be put in some kind of config variable to make them easier to reuse?

**Comment by ewels** [comment]
Sorry, not in the output ([these work as html attributes](https://www.w3schools.com/tags/att_input_minlength.asp) so can just be printed into the HTML template), I more just mean in the Python so that we don't have to keep 3 lots of validation parameters in sync..

**Comment by inaod568** [comment]
@ewels Yeah started looking into that too but I think I'll create a card for this for now. But definitely a good point, should add some config for username, password etc.

**Comment by inaod568** [comment]
@ewels https://github.com/scilifelabdatacentre/dds_web/issues/978

File: dds_web/templates/user/register.html [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
-        Unit name:
-        {{ form.unit_name }}
-        {% endif %}
-        <br>
-        <!-- Submit button -->
-        {{ form.submit }}
+            <!-- Unit -->
+            {% if unit %}
+            Unit name:
+            {{ unit }}
```
Comment (aanil) [inline dds_web/templates/user/register.html:73]: 👍

**Review by aanil — APPROVED** [review]
👍 Looks ok!

**Pull request merged**